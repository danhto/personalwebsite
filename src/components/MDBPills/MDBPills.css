.mdb-tab-content {
    padding-top: 20px;
}