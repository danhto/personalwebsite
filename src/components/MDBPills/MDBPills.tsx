import React, { Component } from "react";
import { MDBContainer, MDBRow, MDBCol, MDBTabPane, MDBTabContent, MDBNav, MDBNavItem, MDBNavLink } from "mdbreact";
import MDBJumbotron from '../MDBJumbotron/MDBJumbotron'

class Pills extends Component {
  state = {
    items: {
      default: "1",
    }
  };

  togglePills = (type, tab) => e => {
    console.log("Toggling pill to state " + tab)
    console.log(this.state)
    console.log(this.state.items['default'] === "3")
    //e.preventDefault();
    if (this.state.items[type] !== tab) {
      let items = { ...this.state.items };
      items[type] = tab;
      this.setState({
        items
      });
    }
  };

  render() {
    return (
      <MDBContainer className="mt-4">
          <MDBRow>
            <MDBCol md="12">
              <h2>All things Dan To</h2>
              <MDBNav className="mt-5 nav-pills">
                <MDBNavItem>
                  <MDBNavLink to="#" active={this.state.items["default"] === "1"} onClick={this.togglePills("default", "1")} >
                    Quick Bio
                  </MDBNavLink>
                </MDBNavItem>
                <MDBNavItem>
                  <MDBNavLink to="#" active={this.state.items["default"] === "2"} onClick={this.togglePills("default", "2")} >
                    Skills
                  </MDBNavLink>
                </MDBNavItem>
                <MDBNavItem>
                  <MDBNavLink to="#" active={this.state.items["default"] === "3"} onClick={this.togglePills("default", "3")} >
                    Interests
                  </MDBNavLink>
                </MDBNavItem>
              </MDBNav>
              <MDBTabContent activeItem={this.state.items["default"]} className="mdb-tab-content">
                <MDBTabPane tabId="1">
                  <MDBJumbotron display={1}/>
                </MDBTabPane>
                <MDBTabPane tabId="2">
                  <MDBJumbotron display={2}/>
                </MDBTabPane>
                <MDBTabPane tabId="3">
                  <p>
                    Est quis nulla laborum officia ad nisi ex nostrud culpa
                    Lorem excepteur aliquip dolor aliqua irure ex. Nulla ut duis
                    ipsum nisi elit fugiat commodo sunt reprehenderit laborum
                    veniam eu veniam. Eiusmod minim exercitation fugiat irure ex
                    labore incididunt do fugiat commodo aliquip sit id deserunt
                    reprehenderit aliquip nostrud. Amet ex cupidatat excepteur
                    aute veniam incididunt mollit cupidatat esse irure officia
                    elit do ipsum ullamco Lorem. Ullamco ut ad minim do mollit
                    labore ipsum laboris ipsum commodo sunt tempor enim
                    incididunt. Commodo quis sunt dolore aliquip aute tempor
                    irure magna enim minim reprehenderit. Ullamco consectetur
                    culpa veniam sint cillum aliqua incididunt velit ullamco
                    sunt ullamco quis quis commodo voluptate. Mollit nulla
                    nostrud adipisicing aliqua cupidatat aliqua pariatur mollit
                    voluptate voluptate consequat non.
                  </p>
                </MDBTabPane>
              </MDBTabContent>
            </MDBCol>
          </MDBRow>
        </MDBContainer>
      );
    }
}

export default Pills;