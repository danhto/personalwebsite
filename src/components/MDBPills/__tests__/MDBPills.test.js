import {configure, shallow} from 'enzyme';
import React from 'react';
import MDBPills from '../MDBPills';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('<MDBPills /> component tests', () => {
    it('expect component to mount', () => {
        const about = shallow(<MDBPills />);
        expect(about).toBeTruthy();
    });
    
    it('match snapshot', () => {
        const about = shallow(<MDBPills />);
        expect(about).toMatchSnapshot();
    })
});