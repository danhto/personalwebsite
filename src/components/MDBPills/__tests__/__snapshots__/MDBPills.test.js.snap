// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MDBPills /> component tests match snapshot 1`] = `ShallowWrapper {}`;
